course/models.py

```diff
@@ -654,12 +654,16 @@ def get_queryset(self):
         return super().get_queryset().select_related(
             'course_instance', 'course_instance__course')
 
-    def get_visible(self, user=None):
+    def get_visible(self, user=None, is_course_staff=False):
         if not user or not user.is_authenticated:
             return self.filter(
                 course_instance__visible_to_students=True,
                 opening_time__lte=timezone.now(),
             )
+        if not user.is_superuser and is_course_staff:
```

Comment:
I think you could just drop the opening time restriction from the branch below. Alternatively, the opening time limitation could probably be moved into that function (i.e. the filter would become `(visible to students && opening time < now) || is_teacher || is_assistant)`)